verifiers/internal/gcb/provenance_test.go

```diff
@@ -0,0 +1,682 @@
+		},
+		{
+			name:     "mismatch fuly qualified digest",
+			path:     "./testdata/gcloud-container-invalid-fullyqualifieddigest.json",
+			hash:     "1a033b002f89ed2b8ea733162497fb70f1a4049a7f8602d6a33682b4ad9921fd",
+			expected: serrors.ErrorMismatchHash,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			content, err := os.ReadFile(tt.path)
+			if err != nil {
+				panic(fmt.Errorf("os.ReadFile: %w", err))
+			}
+
+			prov, err := ProvenanceFromBytes(content)
+			if err != nil {
+				panic(fmt.Errorf("ProvenanceFromBytes: %w", err))
+			}
+
+			if err := setStatement(prov); err != nil {
+				panic(fmt.Errorf("ProvenanceFromBytes: %w", err))
+			}
+
+			provenanceOpts := options.ProvenanceOpts{
+				ExpectedDigest: tt.hash,
+			}
+			err = prov.VerifySummary(&provenanceOpts)
+			if !cmp.Equal(err, tt.expected, cmpopts.EquateErrors()) {
+				t.Errorf(cmp.Diff(err, tt.expected, cmpopts.EquateErrors()))
+			}
+		})
+	}
+}
+
+func Test_VerifyMetadata(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		path     string
+		hash     string
+		expected error
+	}{
+		{
+			name: "valid gcb provenance",
+			path: "./testdata/gcloud-container-github.json",
+			hash: "1a033b002f89ed2b8ea733162497fb70f1a4049a7f8602d6a33682b4ad9921fd",
+		},
+		{
+			name:     "mismatch hash",
+			path:     "./testdata/gcloud-container-github.json",
+			hash:     "2a033b002f89ed2b8ea733162497fb70f1a4049a7f8602d6a33682b4ad9921fd",
+			expected: serrors.ErrorMismatchHash,
+		},
+		{
+			name:     "invalid kind",
+			path:     "./testdata/gcloud-container-invalid-kind.json",
+			hash:     "1a033b002f89ed2b8ea733162497fb70f1a4049a7f8602d6a33682b4ad9921fd",
+			expected: serrors.ErrorInvalidFormat,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			content, err := os.ReadFile(tt.path)
+			if err != nil {
+				panic(fmt.Errorf("os.ReadFile: %w", err))
+			}
+
+			prov, err := ProvenanceFromBytes(content)
+			if err != nil {
+				panic(fmt.Errorf("ProvenanceFromBytes: %w", err))
+			}
+
+			if err := setStatement(prov); err != nil {
+				panic(fmt.Errorf("ProvenanceFromBytes: %w", err))
+			}
+
+			provenanceOpts := options.ProvenanceOpts{
+				ExpectedDigest: tt.hash,
+			}
+			err = prov.VerifyMetadata(&provenanceOpts)
+			if !cmp.Equal(err, tt.expected, cmpopts.EquateErrors()) {
+				t.Errorf(cmp.Diff(err, tt.expected, cmpopts.EquateErrors()))
+			}
+		})
+	}
+}
+
+func Test_VerifyTextProvenance(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		path     string
+		alter    bool
+		expected error
+	}{
+		{
+			name: "valid gcb provenance",
+			path: "./testdata/gcloud-container-github.json",
+		},
+		{
+			name:     "mismatch subject name",
```

Comment:
any way you can also validate same statement header but mismatched provenance? just for safety later in case

Reply:
what do you mean? I validate everything, and also change every single character to be sure it fails (test "mismatch everything"). Let me know if I mis-understood the comment.

Reply:
No! I didn't realize verify was in the loop modifying each char. LGTM